actors/builtin/miner/miner_actor.go

```diff
@@ -874,10 +813,14 @@ func (a Actor) requestTerminatePower(rt Runtime, terminationType power.SectorTer
 	builtin.RequireSuccess(rt, code, "failed to terminate sector power type %v, weights %v", terminationType, weights)
 }
 
-func (a Actor) verifySurprisePost(rt Runtime, st *State, onChainInfo *abi.OnChainSurprisePoStVerifyInfo) {
-	Assert(st.PoStState.isChallenged())
+func (a Actor) verifyWindowedPost(rt Runtime, st *State, onChainInfo *abi.OnChainPoStVerifyInfo) {
 	sectorSize := st.Info.SectorSize
-	challengeEpoch := st.PoStState.SurpriseChallengeEpoch
+
+	// challenge issued at the end (beginning of next?) proving period
+	challengeEpoch := st.PoStState.ProvingPeriodStart + ProvingPeriod
+
+	// TODO: verifying no duplicates here seems wrong, we should be verifying
```

Comment:
👍

Reply:
I'm not so sure @whyrusleeping. I agree we should check exactly what we passed in (but proofs code and randomness check below do that I think). That said, given check for appropriate number of tix, checking there are no duplicates is key. 

Duplicates here does not mean same sector, but same actual challenge. :)